internal/build/buildconfig/manager.go

```diff
@@ -42,18 +50,27 @@ func (bcm *buildConfigManager) Sync(ctx context.Context, mod kmmv1beta1.Module,
 
 		logger.Info("Creating BuildConfig")
 
-		buildConfig, err = bcm.maker.MakeBuildConfig(mod, m, targetKernel, m.ContainerImage, pushImage)
-		if err != nil {
-			return build.Result{}, fmt.Errorf("could not make BuildConfig: %v", err)
-		}
-
-		if err = bcm.client.Create(ctx, buildConfig); err != nil {
-			return build.Result{}, fmt.Errorf("could not create Job: %v", err)
+		if err = bcm.client.Create(ctx, buildConfigTemplate); err != nil {
+			return build.Result{}, fmt.Errorf("could not create BuildConfig: %v", err)
 		}
 
 		return build.Result{Status: build.StatusCreated, Requeue: true}, nil
 	}
 
+	changed, err := bcm.isBuildConfigChanged(buildConfig, buildConfigTemplate)
+	if err != nil {
+		return build.Result{}, fmt.Errorf("could not determine if BuildConfig has changed: %v", err)
+	}
+
+	if changed {
+		logger.Info("The module's build spec has been changed, deleting the current BuildConfig so a new one can be created", "name", buildConfig.Name)
```

Comment:
```suggestion
		logger.Info("The module's build spec has been changed, deleting the current Build resource so a new one can be created", "name", buildConfig.Name)
```

Reply:
Yes, sorry, I though that we are only deleting the build. You can ignore that comment.